supervision/detection/tools/inference_slicer.py

```diff
@@ -68,12 +71,14 @@ def __init__(
         callback: Callable[[np.ndarray], Detections],
         slice_wh: Tuple[int, int] = (320, 320),
         overlap_ratio_wh: Tuple[float, float] = (0.2, 0.2),
-        iou_threshold: Optional[float] = 0.5,
+        overlap_filter: OverlapFilter = OverlapFilter.NON_MAX_SUPPRESSION,
+        iou_threshold: float = 0.5,
```

Comment:
IOU threshold was not optional before - things would _probably_ not work if None was passed. Now I require it.

However, it will be ignored if `OverlapFilter.None` is set.

Reply:
Maybe let's allow people to pass "non_max_suppression` as an argument optionally.